legend-sdlc-server-fs/src/main/java/org/finos/legend/sdlc/server/api/entity/FileSystemEntityApi.java

```diff
@@ -0,0 +1,1118 @@
+                            {
+                                throw new LegendSDLCServerException("File " + fileToRemove + " does not exist");
+                            }
+                            fileToRemove.delete();
+                            git.rm().addFilepattern(commitAction.getFilePath().substring(1)).call();
+                            if (git.status().call().isClean())
+                            {
+                                break;
+                            }
+                            break;
+                        }
+                        default:
+                        {
+                            break;
+                        }
+                    }
+                }
+                RevCommit revCommit = git.commit().setMessage(message).call();
+                repo.close();
+                if (LOGGER.isDebugEnabled())
+                {
+                    LOGGER.debug("Committed {} changes to {}: {}", changeCount, sourceSpecification, revCommit.getId());
+                }
+                return FileSystemRevision.getFileSystemRevision(projectId, branchName);
+            }
+            catch (Exception e)
+            {
+                throw new LegendSDLCServerException("Error occurred while committing changes to workspace " + branchName + " of project " + projectId, e);
+            }
+        }
+
+        private LocalCommitAction fileOperationToCommitAction(ProjectFileOperation fileOperation)
+        {
+            if (fileOperation instanceof ProjectFileOperation.AddFile)
+            {
+                return new LocalCommitAction()
+                        .withAction(LocalCommitAction.Action.CREATE)
+                        .withFilePath(fileOperation.getPath())
+                        .withContent(((ProjectFileOperation.AddFile) fileOperation).getContent());
+            }
+            if (fileOperation instanceof ProjectFileOperation.ModifyFile)
+            {
+                return new LocalCommitAction()
+                        .withAction(LocalCommitAction.Action.UPDATE)
+                        .withFilePath(fileOperation.getPath())
+                        .withContent(((ProjectFileOperation.ModifyFile) fileOperation).getNewContent());
+            }
+            if (fileOperation instanceof ProjectFileOperation.DeleteFile)
+            {
+                return new LocalCommitAction()
+                        .withAction(LocalCommitAction.Action.DELETE)
+                        .withFilePath(fileOperation.getPath());
+            }
+            if (fileOperation instanceof ProjectFileOperation.MoveFile)
+            {
+                throw new LegendSDLCServerException("MOVE operation is not yet supported");
+            }
+            throw new IllegalArgumentException("Unsupported project file operation: " + fileOperation);
+        }
+    }
+
+    public static class AbstractFileSystemFileAccessContext extends AbstractFileAccessContext
```

Comment:
First, since this class is not abstract, its name should not start with "Abstract". That's deceptive.

Second, does it need to be public? If so, it should probably be a class in its own right rather than a nested static class.